src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS/ProjectSystem/VS/LanguageServices/CSharpLanguageServiceHost.cs

```diff
@@ -25,14 +26,11 @@ protected override Guid IntelliSenseProviderGuid
             get { return CSharpIntellisenseProvider; }
         }
 
-        /// <summary>
-        /// Invoked when the UnconfiguredProject is first loaded to initialize language services.
-        /// </summary>
-        [ProjectAutoLoad(ProjectLoadCheckpoint.InitialActiveConfigurationKnown)]
+        [ProjectAutoLoad(ProjectLoadCheckpoint.ProjectFactoryCompleted)]
         [AppliesTo(ProjectCapability.CSharpLanguageService)]
-        private void Initialize()
+        private Task OnProjectFactoryCompleted()
         {
-            var nowait = InitializeAsync();
+            return InitializeAsync();
```

Comment:
I was thinking of InitializeAsync as unknown code as well (if InitializeAsync wasn't doing a ConfigureAwait(false) then there would be a difference in your test). However since that's just our method, you are right, it shouldn't matter here.